### PR TITLE
Add virtualbox as builder

### PR DESCRIPTION
**Pull request by bartjanssens92**
Create a builder called 'archlinux-virtualbox'. This will build a vagrant box for use with virtualbox.
Also updated the version of the iso to use and refactored the ssh.sh script to pre-install.sh.

The pre-install.sh script has the same function as the ssh.sh script but adds some nice verbose output for when not building headless/debugging.

### CODE REVIEW

File: arch-template.json [hunk 1]
```diff
@@ -5,9 +5,9 @@
   "variables": {
     "cloud_token": "{{env `ATLAS_TOKEN`}}",
     "box_version": "{{env `VERSION`}}",
-    "iso_url": "https://archlinux.cu.be/iso/2015.07.01/archlinux-2015.07.01-dual.iso",
-    "iso_checksum": "a025ac8e23131b038a0b7f949faeb04d382e5b25",
-    "iso_checksum_type": "sha1"
+    "iso_url": "/home/bjanssens/Downloads/iso/archlinux-2016.04.01-dual.iso",
```
Comment (LadyNamedLaura) [inline arch-template.json]: This should not EVER get into upsteam.

File: arch-template.json [hunk 2]
```diff
@@ -5,9 +5,9 @@
...
+    "iso_checksum": "56fac9b3eefdf254a53e519dfcb05168",
+    "iso_checksum_type": "md5"
```
Comment (LadyNamedLaura) [inline arch-template.json]: I'd rather keep using the sha1

File: http/pre-install.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+vagrant_passwd='vagrant'
+root_passwd='changeme'
+
+echo "Status: Setting the vagrant root passwd to $root_passwd"
+echo "root:$root_passwd" | chpasswd
+
+echo "Status: Adding the vagrant user"
```
Comment (LadyNamedLaura) [inline http/pre-install.sh]: I don't think we need a vagrant user inside our installer image.

File: http/pre-install.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+echo "Status: Setting the vagrant user passwd to $vagrant_passwd"
+echo "vagrant:$vagrant_passwd" | chpasswd
+
+echo "Status: Configuring sudo for vagrant user"
+cat > /etc/sudoers.d/10_vagrant << EOF
+Defaults:vagrant !requiretty
+  vagrant ALL=(ALL) NOPASSWD: ALL
+EOF
+
+echo "Status: Enableing sshd"
```
Comment (LadyNamedLaura) [inline http/pre-install.sh]: s/enable/start/
say what you do.

File: arch-template.json [hunk 5]
```diff
@@ -38,8 +38,35 @@
...
       ]
+    },
+    {
+      "name": "archlinux-virtualbox",
+      "type": "virtualbox-iso",
+      "headless": true,
+      "guest_os_type": "ArchLinux_64",
+      "iso_url": "/home/bjanssens/Downloads/iso/archlinux-2016.04.01-dual.iso",
+      "iso_checksum": "56fac9b3eefdf254a53e519dfcb05168",
+      "iso_checksum_type": "md5",
```
Comment (LadyNamedLaura) [inline arch-template.json:51]: Use the variables defined at the top.

File: arch-template.json [hunk 6]
```diff
@@ -38,8 +38,35 @@
...
       "boot_command": [
-        "<tab> script=http://{{ .HTTPIP }}:{{ .HTTPPort }}/ssh.sh<enter>"
+        "<tab> script=http://{{ .HTTPIP }}:{{ .HTTPPort }}/pre-install.sh<enter>"
       ]
+    },
+    {
+      "name": "archlinux-virtualbox",
+      "type": "virtualbox-iso",
+      "headless": true,
+      "guest_os_type": "ArchLinux_64",
```
Comment (LadyNamedLaura) [inline arch-template.json:48]: Does "ArchLinux_64" exist as an explicit type in virtualbox nowadays?

File: arch-template.json [hunk 7]
```diff
@@ -38,8 +38,35 @@
       "http_directory": "http",
       "boot_wait": "2s",
       "boot_command": [
-        "<tab> script=http://{{ .HTTPIP }}:{{ .HTTPPort }}/ssh.sh<enter>"
+        "<tab> script=http://{{ .HTTPIP }}:{{ .HTTPPort }}/pre-install.sh<enter>"
```
Comment (LadyNamedLaura) [inline arch-template.json:41]: If you change to pre-install.sh, also remove the unused ssh.sh

**Comment by LadyNamedLaura** [comment]
a625cb2 should be squashed into 77d0120
changing from ssh.sh to pre-install.sh should be in a separate commit or squashed into 6689bf0

File: arch-template.json [hunk 8]
```diff
@@ -78,6 +78,9 @@
       }, {
         "type": "vagrant-cloud",
         "box_tag": "SimonPe/archlinux",
+        "only": [
+          "archlinux-kvm"
+        ],
```
Comment (LadyNamedLaura) [inline arch-template.json:83]: Motivation?

File: arch-template.json (hunk 6 above)
Comment (bartjanssens92) [inline arch-template.json:48]: Yes it does:

```
vboxmanage list ostypes
...
ID:          ArchLinux
Description: Arch Linux (32-bit)
Family ID:   Linux
Family Desc: Linux
64 bit:      false

ID:          ArchLinux_64
Description: Arch Linux (64-bit)
Family ID:   Linux
Family Desc: Linux
64 bit:      true 
```

File: arch-template.json (hunk 8 above)
Comment (bartjanssens92) [inline arch-template.json:83]: As I will be making lot's of changes to the archlinux-virtualbox it isn't quite ready to be published to vagrant-cloud yet.

**Comment by bartjanssens92** [comment]
Going to fix these issues and create a new pull request on a separate branch.

**Pull request closed without merging**